geode-gfsh/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/StatusRedundancyCommandDUnitTest.java

```diff
@@ -0,0 +1,281 @@
+  private static final String UNSATISFIED_REGION = "unsatisfiedRedundancy";
+  private static final int UNSATISFIABLE_COPIES = SERVERS_TO_START;
+  private static final String NO_CONFIGURED_REDUNDANCY_REGION = "noConfiguredRedundancy";
+  private static final String ZERO_COPIES_REGION = "zeroRedundantCopies";
+  private static final List<String> regionNames = Arrays.asList(SATISFIED_REGION,
+      UNSATISFIED_REGION, NO_CONFIGURED_REDUNDANCY_REGION, ZERO_COPIES_REGION);
+
+  @Before
+  public void setUp() throws Exception {
+    servers = new ArrayList<>();
+    locator = cluster.startLocatorVM(0);
+    locatorPort = locator.getPort();
+    IntStream.range(0, SERVERS_TO_START)
+        .forEach(i -> servers.add(cluster.startServerVM(i + 1, locatorPort)));
+    gfsh.connectAndVerify(locator);
+  }
+
+  @Test
+  public void statusRedundancyWithNoArgumentsReturnsStatusForAllRegions() {
+    createAndPopulateRegions();
+
+    String command = new CommandStringBuilder(COMMAND_NAME).getCommandString();
+
+    CommandResultAssert commandResult = gfsh.executeAndAssertThat(command);
+    commandResult.statusIsSuccess();
+
+    verifySummarySection(commandResult, 1, 1, 2);
+
+    InfoResultModelAssert zeroRedundancy = commandResult.hasInfoSection(ZERO_REDUNDANCY_SECTION);
+    zeroRedundancy.hasHeader().isEqualTo(NO_REDUNDANT_COPIES_FOR_REGIONS);
+    zeroRedundancy.hasOutput().contains(ZERO_COPIES_REGION);
+
+    InfoResultModelAssert underRedundancy = commandResult.hasInfoSection(UNDER_REDUNDANCY_SECTION);
+    underRedundancy.hasHeader().isEqualTo(REDUNDANCY_NOT_SATISFIED_FOR_REGIONS);
+    underRedundancy.hasOutput().contains(UNSATISFIED_REGION);
+
+    InfoResultModelAssert satisfiedRedundancy =
+        commandResult.hasInfoSection(SATISFIED_REDUNDANCY_SECTION);
+    satisfiedRedundancy.hasHeader().isEqualTo(REDUNDANCY_SATISFIED_FOR_REGIONS);
+    satisfiedRedundancy.hasOutput().contains(SATISFIED_REGION,
+        NO_CONFIGURED_REDUNDANCY_REGION);
+  }
+
+  @Test
+  public void statusRedundancyWithIncludeRegionArgumentReturnsStatusForOnlyThatRegion() {
+    createAndPopulateRegions();
+
+    String command = new CommandStringBuilder(COMMAND_NAME)
+        .addOption(INCLUDE_REGION, SATISFIED_REGION)
+        .getCommandString();
+
+    CommandResultAssert commandResult = gfsh.executeAndAssertThat(command);
+    commandResult.statusIsSuccess()
+        .hasNoSection(ZERO_REDUNDANCY_SECTION)
+        .hasNoSection(UNDER_REDUNDANCY_SECTION)
+        .doesNotContainOutput(UNSATISFIED_REGION)
+        .doesNotContainOutput(ZERO_COPIES_REGION)
+        .doesNotContainOutput(NO_CONFIGURED_REDUNDANCY_REGION);
+
+    verifySummarySection(commandResult, 0, 0, 1);
+
+    InfoResultModelAssert satisfiedRedundancy =
+        commandResult.hasInfoSection(SATISFIED_REDUNDANCY_SECTION);
+    satisfiedRedundancy.hasHeader().isEqualTo(REDUNDANCY_SATISFIED_FOR_REGIONS);
+    satisfiedRedundancy.hasOutput().contains(SATISFIED_REGION);
```

Comment:
This code block is duplicated within the test, it can be extracted to a separate method.